test/hotspot/jtreg/compiler/loopopts/parallel_iv/TestParallelIvInIntCountedLoop.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts.parallel_iv;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+import java.util.Random;
+import java.util.function.Function;
+
+/**
+ * @test
+ * @bug 8328528
+ * @summary test the long typed parallel iv replacing transformation for int counted loop
+ * @library /test/lib /
+ * @run driver compiler.loopopts.parallel_iv.TestParallelIvInIntCountedLoop
+ */
+public class TestParallelIvInIntCountedLoop {
+    private static final int stride;
+    private static final int stride2;
+
+    static {
+        // stride2 must be a multiple of stride and must not overflow for the optimization to work
+        stride = new Random().nextInt(1, Integer.MAX_VALUE / 16);
+        stride2 = stride * new Random().nextInt(1, 16);
+    }
```

Comment:
So `stride` and `stride2` are compile-time constants. That is intended, right?

Reply:
Yes, this is intended. I followed [your suggestion](https://github.com/openjdk/jdk/pull/18489#discussion_r1638725652) to expand testing with some random numbers. Please let me know if I misunderstood.